app/models.py

```diff
@@ -1513,6 +1517,47 @@ def update_from_json(self, data):
         self.data = new_data
 
 
+class CaseStudy(db.Model):
+    __tablename__ = 'case_study'
+
+    id = db.Column(db.Integer, primary_key=True)
+    data = db.Column(JSON, nullable=False)
+    supplier_code = db.Column(db.BigInteger, db.ForeignKey('supplier.code'), nullable=False)
+    created_at = db.Column(db.DateTime, index=True, nullable=False, default=datetime.utcnow)
+
+    supplier = db.relationship('Supplier', lazy='joined')
+
+    @validates('data')
+    def validates_data(self, key, data):
+        data = drop_foreign_fields(data, [
+            'supplierCode'
+        ])
+        data = strip_whitespace_from_data(data)
+        data = purge_nulls_from_data(data)
+
+        return data
+
+    def serialize(self):
+        data = self.data.copy()
+        data.update({
+            'id': self.id,
+            'supplierCode': self.supplier_code,
+            'supplierName': self.supplier.name,
+            'createdAt': self.created_at.strftime(DATETIME_FORMAT),
+            'links': {
+                'self': url_for('.get_work_order', work_order_id=self.id),
+                'supplier': url_for(".get_supplier", code=self.supplier_code),
+            }
+        })
+
+        return data
+
+    def update_from_json(self, data):
```

Comment:
There is a useful sqlalchemy extension that prevents you from having to do this copying dance every time you modify your json: http://docs.sqlalchemy.org/en/latest/orm/extensions/mutable.html